Remove usage of s3-encrypted

MalinAhlberg opened the pull request:
**Related issue(s) and PR(s)**  
This PR closes #497 .


**Description**
-  Usage and references to `/s3-encrypted` are replaced with `/s3`.
- The latest tag will be used for the image of download.
- The endpoint `/files` will not be used anymore, as it is for encrypted data
- Support for removing the user name part of the filepath is improved (but can be removed when we merge [`feature/do-not-leak-user-ids`](https://github.com/neicnordic/sensitive-data-archive/tree/feature/do-not-leak-user-ids) )
- Encrypted files are expected
- Tests are updated 

**How to test**
Use `sda-cli` to dowload files.

**To do**
- [ ] Make a release to v0.1.4

nanjiangshu commented on download/download.go:
```diff
@@ -510,3 +507,30 @@ func GetURLsFile(urlsFilePath string) (urlsList []string, err error) {
 	return urlsList, scanner.Err()
 }
+
+func AnonymizeFilepath(filePath string) string {
```
Nice refactoring! If this function is supposed to be used by other modules as well, I think it is better to be placed in `helpers.go`. 
And would be even better to add a comment to explain that it is trying to remove email like userID from the filepath. Maybe add a unit text as well?

MalinAhlberg replied:
Fixed in 3a00170

nanjiangshu commented on download/download.go:
```diff
@@ -510,3 +507,30 @@ func GetURLsFile(urlsFilePath string) (urlsList []string, err error) {
+	return filePath
+}
+
+func getPublicKey64() (string, error) {
```
This function looks quite generic, so maybe better to be placed in `helpers.go`.

MalinAhlberg replied:
Fixed in 3a00170

nanjiangshu reviewed (commented):
Looks great! I just added two minor comments regarding the placement of the functions and also potential unit tests.

MalinAhlberg commented:
Thank you, @nanjiangshu :+1: ! I addressed your suggestions in 3a00170. I did not add any unit test for `AnonymizeFilepath`, though, as I believe it will not be used much in the future, since we are moving away from storing userids as part of the filepath.
